src/libsync/vfs/cfapi/vfs_cfapi.cpp

```diff
@@ -308,6 +309,15 @@ void VfsCfApi::requestHydration(const QString &requestId, const QString &path)
     // of the placeholder with decrypted data
     if (record._isE2eEncrypted || !record._e2eMangledName.isEmpty()) {
         qCInfo(lcCfApi) << "Couldn't hydrate, the file is E2EE this is not supported";
+
+        QMessageBox e2eeFileDownloadRequestWarningMsgBox(QMessageBox::Warning,
+            "Download of End-to-End encrypted file failed",
+            "It seems that you are trying to download a virtual file that"
+            " is End-to-End encrypted. Implicitly downloading such files is not"
+            " supported at the moment. To workaround that issue, go to the"
+            " settings and mark the encrypted folder with \"Make always available locally\".");
+        e2eeFileDownloadRequestWarningMsgBox.exec();
```

Comment:
Better use one of the static methods if you do this.

https://doc.qt.io/qt-5/qmessagebox.html#warning

Reply:
But then I have to pass a parent widget, and I don't have one in this place.

Reply:
Just pass nullptr :-)

You'd have to pass one with the instance too, you just benefit from the `= nullptr` on that parameter.

Reply:
Okay. While rereading the documentation, I have seen that the explaining text should be set with `setInformativeText()`. Seems like that is not possible with the static methods.

Reply:
Well, I'd expect what you pass to setText to be the title and what you pass to setInformativeText to be the text.

Reply:
But as far as I understand the documentation, the title should be just the application name to be conform with Windows and macOS human interface guidelines.

Reply:
Not uncommon to have more or something else in there, but OK, maybe. I admit I didn't check their HIG in a while.

Reply:
If this is really the case and the static method suggested by @er-vin indeed is incomplete for this scenario, we may consider implementing our own static method that `does it all` with QMessageBox. Just to create tools we can reuse in these 2 places where the message box is needed, and, also in the future. Just a suggestion of possible compromise.

Reply:
That is a good idea @allexzander , but I'm not sure where to create it. One message box is part libsync, the other of gui. Then there is the common folder, but in this folder but there is a comment that it should compile without gui. Any suggestions?

Reply:
@FlexW Within the `guiutility.h` for example.

Reply:
@allexzander but then it not usable for one of the dialogs, because one of the dialogs is in libsync. maybe the architecture can be changed, that both dialogs get created within gui.

Reply:
> @allexzander but then it not usable for one of the dialogs, because one of the dialogs is in libsync. maybe the architecture can be changed, that both dialogs get created within gui.

@FlexW Alrighty then. I wouldn't hurry with this helper function in this case.